File: 백준/Gold/9576. 책 나눠주기/책 나눠주기.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"sort"
)

var (
	reader = bufio.NewReader(os.Stdin)
	writer = bufio.NewWriter(os.Stdout)
)

func main() {
	defer writer.Flush()
	var tc int
	fmt.Fscanf(reader, "%d\n", &tc)
	for i := 0; i < tc; i++ {
		var n, m, cnt int
		fmt.Fscanf(reader, "%d %d\n", &n, &m)
		var books = make([]bool, n+1)
		var req = make([][2]int, m)
		for j := 0; j < m; j++ {
			fmt.Fscanf(reader, "%d %d\n", &req[j][0], &req[j][1])
		}
		sort.SliceStable(req, func(i, j int) bool {
			return req[i][1] < req[j][1]
		})
		for j := 0; j < len(req); j++ {
			for k := req[j][0]; k <= req[j][1]; k++ {
				if books[k] == false {
					cnt++
					books[k] = true
					break
				}
			}
		}
		fmt.Fprintf(writer, "%d\n", cnt)
	}
}

File: 백준/Gold/9576. 책 나눠주기/책 나눠주기.py
import sys 
input = sys.stdin.readline
sys.setrecursionlimit(10**6)
tc = int(input())
for _ in range(tc):
    n, m = map(int, input().split())
    books = [False for _ in range(n+1)]
    req = []
    for _ in range(m):
        req.append(list(map(int, input().split())))
    req.sort(key=lambda x: x[1])
    cnt = 0
    for a, b in req:
        for i in range(a, b+1):
            if not books[i]:
                cnt += 1
                books[i] = True
                break
    print(cnt)